src/Simplex/Messaging/Agent/NtfSubSupervisor.hs

```diff
@@ -91,31 +101,72 @@ processNtfSub c (rcvQueue@RcvQueue {server = smpServer, rcvId}, cmd) = do
         Just (doWork, _) -> void . atomically $ tryPutTMVar doWork ()
 
 runNtfWorker :: AgentMonad m => AgentClient -> NtfServer -> TMVar () -> m ()
-runNtfWorker _c srv doWork = forever $ do
+runNtfWorker c srv doWork = forever $ do
   void . atomically $ readTMVar doWork
-  withStore $ \st ->
-    getNextNtfSubAction st srv >>= \case
-      Nothing -> void . atomically $ tryTakeTMVar doWork
-      Just (_sub, ntfSubAction) ->
-        forM_ ntfSubAction $ \case
-          NSANew _nKey -> pure ()
-          NSACheck -> pure ()
-          NSADelete -> pure ()
+  ntfToken_ <- getNtfToken
+  nextSubAction <- withStore (`getNextNtfSubAction` srv)
+  case (ntfToken_, nextSubAction) of
+    (Nothing, _) -> noWorkToDo
+    (Just NtfToken {ntfTokenId = Nothing}, _) -> noWorkToDo
+    (Just _, Nothing) -> noWorkToDo
```

Comment:
it does seem like the return from store method should be Maybe of whatever, to have Nothing always mean "no work", and not three different values...